Create guide/overview for the SIG Security External Audit subproject (#5870)
kubernetes/community: sig-security/sig-security-external-audit/README.md

# SIG Security External Audit Subproject

## Overview

The SIG Security External Audit subproject (subproject, henceforth) is responsible for coordinating regular, comprehensive, third-party security audits.
The subproject publishes the deliverables of the audit after abiding to the [Security Release Process](https://github.com/kubernetes/security/blob/master/security-release-process.md) and [embargo policy](https://github.com/kubernetes/security/blob/master/private-distributors-list.md#embargo-policy).

  - [Request for Proposal (RFP)](#rfp)
    - [Security Audit Scope](#security-audit-scope)
    - [Vendor and Community Questions](#vendor-and-community-questions)
  - [Review of Proposals](#review-of-proposals)
  - [Vendor Selection](#vendor-selection)
  - [Deliverables](#deliverables)

## RFP

> The subproject produces a RFP for a third-party, comprehensive security audit. The subproject publishes the RFP in the `kubernetes/community/sig-security` repository. The subproject defines the scope, schedule, methodology, selection criteria, and deliverables in the RFP.

Reviewer: I don't think sig-security is its own repo. just a folder

```suggestion
The subproject produces a RFP for a third-party, comprehensive security audit. The subproject publishes the RFP in the 'sig-security' folder in the `kubernetes/community/` repository. The subproject defines the scope, schedule, methodology, selection criteria, and deliverables in the RFP.
```

Author: applied suggestion in commit 947b473

Previous RFPs:
  - [2019](https://github.com/kubernetes/community/blob/master/sig-security/security-audit-2019/RFP.md)
  - [2021](https://github.com/kubernetes/community/blob/master/sig-security/security-audit-2021/RFP.md)

> As efforts begin for the year's security audit, create a tracking issue for the security audit in `kubernetes/community` with the `sig/security` label.

Reviewer 2: Would it make sense to add another label for this so you don't have to sift through all the other sig/security things to find the ones that pertain specifically to the audit?

Author: @IanColdwater, good point, how about adding a `/label external-audit`

Author: looks like `/label external-audit` is not allowed

Proposal reviewers are members of a private Google group and private Slack channel to exchange sensitive, confidential information and to share artifacts.

### Conflict of Interest

There is a possibility of a conflict of interest between a proposal reviewer and a vendor. Proposal reviewers should not have a conflict of interest. Examples of conflict of interest:
  - Proposal reviewer is employed by a vendor who submitted a proposal
  - Proposal reviewer has financial interest directly tied to the audit

Should a conflict arise during the proposal review, reviewers should notify the subproject owner and SIG Security chairs when they become aware of the conflict.

> The _Conflict of Interest_ section is inspired by the [CNCF Security TAG security reviewer process](https://github.com/cncf/tag-security/blob/main/assessments/guide/security-reviewer.md#conflict-of-interest).

## Vendor Selection

> On the vendor selection date, the subproject will publish a the selected vendor in the `kubernetes/community/sig-security` repository. [Example from the 2019 audit](https://github.com/kubernetes/community/blob/master/sig-security/security-audit-2019/RFP_Decision.md).

Reviewer: ```suggestion
On the vendor selection date, the subproject will publish a the selected vendor in the 'sig-security' folder in the `kubernetes/community` repository. [Example from the 2019 audit](https://github.com/kubernetes/community/blob/master/sig-security/security-audit-2019/RFP_Decision.md).
```

Author: applied suggestion in commit 947b473

## Deliverables

> The deliverables of the audit are defined in the RFP e.g. findings report, threat model, white paper, audited reference architecture spec (with yaml manifests) and published in the `kubernetes/community/sig-security` repository. [Example from the 2019 audit](https://github.com/kubernetes/community/tree/master/sig-security/security-audit-2019/findings)

Reviewer: ```suggestion
The deliverables of the audit are defined in the RFP e.g. findings report, threat model, white paper, audited reference architecture spec (with yaml manifests) and published in the 'sig-security' folder in the `kubernetes/community` repository. [Example from the 2019 audit](https://github.com/kubernetes/community/tree/master/sig-security/security-audit-2019/findings)
```

Author: applied suggestion in commit 947b473
